### PR TITLE
Fix flaky jobset integration test

### LINKED ISSUE #2398: Flaky integration tests for jobset controller
**What happened**:
The integration test failed

**What you expected to happen**:
Not to fail

**How to reproduce it (as minimally and precisely as possible)**:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2363/pull-kueue-test-integration-main/1800218473264058368

**Anything else we need to know?**:

**Environment**:

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind flake

#### What this PR does / why we need it:
Fixes the flaky test of jobset controller 
#### Which issue(s) this PR fixes:

Fixes #2398 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 1]
```diff
@@ -344,7 +344,7 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
 				createdJobSet1 := &jobsetapi.JobSet{}
 				gomega.Eventually(func(g gomega.Gomega) {
 					g.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
-					g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, false)).To(gomega.BeTrue())
+					g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, true)).To(gomega.BeFalse())
```
Comment (mbobrovskyi) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, false)).To(gomega.BeFalse())

Comment (mbobrovskyi) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:347]:
```suggestion
					g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, false)).To(gomega.BeFalse())
```

Comment (mszadkow) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: I think the default value should be the reverse of what we expect, wdyt?

Comment (mbobrovskyi) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: For my opinion it should be default value.

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 2]
```diff
@@ -344,7 +344,7 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
 				createdJobSet1 := &jobsetapi.JobSet{}
 				gomega.Eventually(func(g gomega.Gomega) {
 					g.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
-					g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, false)).To(gomega.BeTrue())
```
[the remark this answers is not in the document]
Comment (mimowo) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: Ah, my mistake, thanks for explaining!
I'm fine with the change, but yet wondering if there is a way to express the check on a ptr field, expecting some concrete value. I find the check `.To(gomega.BeFalse())` tricky to parse quickly.

Wdyt about `g.Expect(createdJobSet.Spec.Suspend).Should(gomega.BeEquivalentTo(ptr.To(false)))`?
Or another approach, where we test the value explicitly, rather than by negating.

File: test/integration/controller/jobs/jobset/jobset_controller_test.go (hunk 1 above)
Comment (trasc) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:347]:
```suggestion
					g.Expect(ptr.Deref(createdJobSet1.Spec.Suspend, false)).To(gomega.BeFalse())
```

File: test/integration/controller/jobs/jobset/jobset_controller_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mszadkow) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: We could try this, but IMHO it's the same level of unpacking the meaning.
We collectively think it's fine to keep it, though I was explained by @trasc that in fact def value should be false - https://github.com/kubernetes-sigs/kueue/pull/2406#discussion_r1637906254

[the remark this answers is not in the document]
Comment (mimowo) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: The two methods differ slightly. In the current form the check accepts `.spec.suspend=nil`, which is not expected to be set by Kueue. 

OTOH, it is a corner case. If you are aware of the difference and still think you like the check, I'm ok to merge it.

[the remark this answers is not in the document]
Comment (mimowo) [reply test/integration/controller/jobs/jobset/jobset_controller_test.go:347]: synced with @mszadkow and we will keep as is for consistency with the rest of the codebase. We can consider a pr which updates it across codebase, but we can park the idea for now.
